hangman.py

```diff
@@ -0,0 +1,356 @@
+                print("Oops! You have no warnings left, so you lose ine guess:",end=' ')
+                get_guessed_word(secret_word,letters_guessed)
+        else:
+            if letter in secret_word:
+                letters_guessed.append(letter)
+                print("Good guess:", get_guessed_word(secret_word, letters_guessed))
+            else:
+                letters_guessed.append(letter)
+                if letter in "aeoiu":
+                    guesses_point-=2
+                else:
+                    guesses_point-=1
+                print("Oops! That letter is not in my word:", get_guessed_word(secret_word,letters_guessed))
+        if is_word_guessed(secret_word,letters_guessed)==0:
+            final_points=guesses_point*int(len(secret_word))
+            print("Congratulations, you won!")
+            print("Your total score for this game is:",final_points)
+            guesses_point=-1
+        if guesses_point==0:
+            print("-------")
+            print("Sorry, you ran out of guessed. The word was", secret_word)
+            break
+
+
+
+# When you've completed your hangman function, scroll down to the bottom
+# of the file and uncomment the first two lines to test
+#(hint: you might want to pick your own
+# secret_word while you're doing your own testing)
+
+
+# -----------------------------------
+
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the 
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise: 
+    '''
+    my_word1 = "".join(my_word.split(" "))
+    if len(my_word1) == len(other_word):
+        for i in range(len(my_word1)):
+            if my_word1[i] == '_':
+                continue
+            elif my_word1[i] == other_word[i]:
+                continue
+            else:
+                return False
+        return True
+    else:
+        return False
+
+
+
+def show_possible_matches(my_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    returns: nothing, but should print out every word in wordlist that matches my_word
+             Keep in mind that in hangman when a letter is guessed, all the positions
+             at which that letter occurs in the secret word are revealed.
+             Therefore, the hidden letter(_ ) cannot be one of the letters in the word
+             that has already been revealed.
+
+    '''
+    final_list = []
+    for i in list(wordlist):
+        if match_with_gaps(my_word, i) == True:
+            final_list.append(i)
+        else:
+            continue
+    if len(final_list) == 0:
+        print("No mathces found")
+    else:
+        return final_list
+
+
+
+
+def hangman_with_hints(secret_word):
```

Comment:
Same comment as for `hangman` function, but also this function is a copy-paste of `hangman` function. Try to avoid code duplication. Try to combine these two functions togather